boards/nxp/mimxrt1050_evk/board.yml

```diff
@@ -0,0 +1,12 @@
+board:
+  name: mimxrt1050_evk
+  vendor: nxp
+  socs:
+    - name: rt1052
+  revision:
+    format: "custom"
+    default: "hyperflash"
+    revisions:
+      - name: "hyperflash"
+      - name: "qspi"
```

Comment:
seems these should be variants not revisions?

Reply:
Not only are revisions the most convenient way to handle this in the board definition, but also, in my view, these really are different board revisions, because switching between hyperflash and qspi requires reworking the board hardware with a soldering iron. And hyperflash and qspi are not really variants of the SOC at all, they are still using the same interface to the SOC, they are just two different flash chips. The fact they are just on the same board is just because it's an EVK... these are really two different board versions that happen to exist on the same PCB.